packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -42,6 +45,44 @@ export interface MonacoModelContentChangedEvent {
     readonly contentChanges: TextDocumentContentChangeEvent[];
 }
 
+export const MonacoEditorModelFactoryHandler = Symbol('MonacoEditorModelFactoryHandler');
+export interface MonacoEditorModelFactoryHandler {
+
+    canHandle(resource: Resource): MaybePromise<number>;
+
+    createModel(
+        resource: Resource,
+        m2p: MonacoToProtocolConverter,
+        p2m: ProtocolToMonacoConverter,
+        options?: { encoding?: string | undefined }
+    ): MaybePromise<MonacoEditorModel>;
+
+}
+
+@injectable()
+export class MonacoEditorModelFactory {
```

Comment:
Have you seen `MonacoTextModelService.registerTextModelContentProvider`? Should not we rather implement it instead of adding new concepts?

Also it should be inlined into  `MonacoTextModelService`, only this service should be used to create models, nothing else. `MonacoEditorModelFactoryHandler` then could be renamed to `MonacoEditorModelFactory`.

Reply:
> Have you seen `MonacoTextModelService.registerTextModelContentProvider`? Should not we rather implement it instead of adding new concepts?

I do not get this, the current implementation is a NOOP. How should this work?

Reply:
I could not implement it without two distinct ~contributions~ factories; one for the mode, the other for the editor. Let me know what you think.

Reply:
ok, let's go with a model factory